Applications/TextEditor/TextEditorWidget.cpp

```diff
@@ -33,17 +39,33 @@ TextEditorWidget::TextEditorWidget()
     });
 
     auto open_action = GAction::create("Open document", { Mod_Ctrl, Key_O }, GraphicsBitmap::load_from_file("/res/icons/16x16/open.png"), [this](const GAction&) {
-        GFilePicker picker;
-
-        if (picker.exec() == GDialog::ExecOK) {
-            m_path = picker.selected_file().string();
+        Optional<String> open_name = GFilePicker::get_open_filepath();
+        if (open_name.has_value() && !open_name.value().is_null()) {
+            m_path = open_name.value();
             open_sesame(m_path);
         }
     });
 
     auto save_action = GAction::create("Save document", { Mod_Ctrl, Key_S }, GraphicsBitmap::load_from_file("/res/icons/16x16/save.png"), [this](const GAction&) {
-        dbgprintf("Writing document to '%s'\n", m_path.characters());
-        m_editor->write_to_file(m_path);
+        if (!m_path.is_empty()) {
+            if (!m_editor->write_to_file(m_path)) //Prettry redundant code here. not sure how I wanna approach it.
+                GMessageBox::show("Unable to save file.\n", "Error", GMessageBox::Type::Error, window());
+        } else {
+            Optional<String> open_name = GFilePicker::get_save_filepath();
+            if (open_name.has_value() && !open_name.value().is_null()) {
```

Comment:
`get_save_filepath()` should return an `Optional<String>`. Then you don't need to check `is_null()`.
Also the variable name should probably be `save_name` instead of `open_name` ;)